ironfish-rust/src/transaction/unsigned.rs

```diff
@@ -256,6 +256,36 @@ impl UnsignedTransaction {
         Ok(transaction)
     }
 
+    pub fn add_signature(&mut self, signature: [u8; 64]) -> Result<Transaction, IronfishError> {
```

Comment:
We might be able to refactor `aggregate_signature_shares` to call `add_signature`